### ISSUE TITLE
logging function as in microsoft Notepad

### ISSUE THREAD

**Issue by Olifair** [notepad-plus-plus/notepad-plus-plus#5537]
### Description of the Issue
Microsoft Notepad has a function where : 
 * when a file starts with ".LOG"
 * everytime the file is opened 
 * the date and time are appended to the end of the file in a new line
it would be great if NotePad++ would also provide this functionality available in the simple Microsoft Notepad.
Thanks in advance for considering this
Cheers

**Comment by MetaChuh** [comment]
@Olifair 

many thanks for your feature idea.

while your request can be a nice add on idea for a plugin, it will currently not be implemented directly into the notepad++ core, as it does not seem suitable for the majority of notepad++ users.

such feature would automatically alter a log file upon loading, possibly causing an accidental, unwanted saving of a log file which should remain in its original state.

thanks and best regards.

-----
```
general notes:

please close your issues as soon as you are satisfied, to help
keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

please also conduct a search at the issue tracker beforehand, to check if a similar issue
already exists.

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and
search if related topics exists. you are welcome to post either at similar
topics, or to create a new topic at Help Wanted or General Discussion.
(no extra account is needed, just use your github account to sign in)
```
[<p align="center"><kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd></p>](https://notepad-plus-plus.org/community/)

**Comment by Olifair** [comment]
MetaChuh : thanks for your rapid response. I think there is a misunderstanding : the functionality in Notepad does not use a log file, that is a file with the extension .log e.g. "myfile.log", but a file with a text extension (e.g. "myfile.txt") that has as a first line ".LOG". So this does not interfere with log file reading, which would not be a good thing, I agree. I don't think that  the ".LOG" feature (adding a date stamp at each opening) would interfere with a majority of users, I have actually found quite a few comments in different forums where different NotePad++ users had asked if this feature would also be working in NotePad++. But I quite understand that this feature might rather be something for a plugin.  Cheers.

**Comment by sasumner** [comment]
@Olifair You can certainly get this functionality via a simple Pythonscript, e.g.:

    from datetime import datetime

    def callback_npp_FILEOPENED(args):
        if editor.getTextLength() >= 4:
            if editor.getLine(0).startswith('.LOG'):
                line_ending = [ '\r\n', '\r', '\n' ][notepad.getFormatType()]
                now = datetime.now()
                time_str = now.strftime("%m/%d/%Y, %H:%M:%S")
                editor.appendText(line_ending + time_str + line_ending)

    notepad.callback(callback_npp_FILEOPENED, [NOTIFICATION.FILEOPENED])

**Comment by Olifair** [comment]
@sasumner Thanks for the script. But how do I use this script ? How to transfrom this to a NotePad++ Plugin ?  I don't think that I have Python installed on my Windows 10 laptop.

**Comment by Olifair** [comment]
@sasumner  Ok, found Pythonscript. Is this what I could use ? However seems only to run with 32bit, as luck wills I have 64bit version installed. . :-(

**Comment by MetaChuh** [comment]
@Olifair 

here's the [Guide: How to install the PythonScript plugin on Notepad++ 7.6.3, 7.6.4 and above](https://notepad-plus-plus.org/community/topic/17256/guide-how-to-install-the-pythonscript-plugin-on-notepad-7-6-3-7-6-4-and-above/1) for both 32 and 64 bit.

**Comment by Olifair** [comment]
@MetaChuh : sorry, I had edited my comment above that I had found a script to install and that this had worked fine. 
@sasumner Actually I would be quite content to simply be able to insert the date. Therefore I have installed Phythonscript and based on your script above and the example scripts created the following script : 
import datetime
now = datetime.now()
time_str = now.strftime("%m/%d/%Y, %H:%M:%S")
editor.addText(time_str)

but this does not seem to work (trying to run it by going into the Extensions/Phythonscript/Scripts menu and clicking on the scripts name. 
Even a simple script such as 
Editor.addText("Test")
does not seem to have any effect. 
Any ideas as to why ? 
Thanks in advance
Oliver
